spec/unit/shoulda/matchers/active_model/validate_confirmation_of_matcher_spec.rb

```diff
@@ -146,4 +146,36 @@
   def validation_matcher_scenario_args
     super.deep_merge(matcher_name: :validate_confirmation_of)
   end
+
+  context 'when attribute to confirm is type integer' do
+    context 'when the model has a confirmation validation' do
+      it 'passes' do
+         builder = builder_for_record_validating_confirmation(type: :integer)
+      expect(builder.record).
+        to validate_confirmation_of(builder.attribute_to_confirm)
+      end
+    end
+
+    context 'when the model has all attributes, but does not have the confirmation'do
```

Comment:
Let's tweak this wording a bit. What do you think about just using:

* "when the model has a confirmation validation on the given attribute"
* "when the model does not have a confirmation validation on the given attribute"